src/vcpkg/commands.integrate.cpp

```diff
@@ -336,9 +328,16 @@ namespace vcpkg::Commands::Integrate
                           create_appdata_shortcut(paths.buildsystems_msbuild_targets),
                           VCPKG_LINE_INFO);
 
-        integrate_install_msbuild14(fs);
+        install_complete = integrate_install_msbuild14(fs);
 #endif
-        msg::println(Color::success, msgAppliedUserIntegration);
+        if (install_complete)
+        {
+            msg::println(Color::success, msgAppliedUserIntegration);
+        }
+        else
+        {
+            msg::println_warning(msgIntegrationFailed);
```

Comment:
This change also causes `integrate install` to exit with a 0 exit code (SUCCESS) when the msbuild integration fails.

I think you will need to check `install_complete` below and conditionalize the `exit_success`.